MSAL/src/MSALDeviceInformation+Internal.h

```diff
@@ -31,7 +31,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 @property (nonatomic, readwrite) MSALDeviceMode deviceMode;
 
-- (instancetype)initWithMSIDDeviceInfo:(MSIDDeviceInfo *)deviceInfo;
+- (instancetype)init;
+- (void) addExtraDeviceInformation:(MSIDDeviceInfo *)deviceInfo;
```

Comment:
Can we keep deviceInfo as a required parameter? We shouldn't be creating this class without device info.

Reply:
We are setting DeviceMode from DeviceInfo and that member is available only on iOS 13.0+.
If we add DeviceInfo as must have filed then we may not able to add WPJ device registration info(deviceID, UPN etc)

Reply:
As discussed, Keeping DeviceInfo initialiser as it is.